### PR TITLE
Kernel: In Mutexes, don't check that interrupts are enabled during early boot

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
The interrupts enabled check in the Kernel mutex is there so that we don't lock mutexes within a spinlock, because mutexes reenable interrupts and that will mess up the spinlock in more ways than one if the thread moves processors. This check is guarded behind a debug flag because it's too hard to fix all the problems at once, but we regressed and weren't even getting to init stage 2 with it enabled. With this commit, we get to stage 2 again. In early boot, there are no interrupts enabled and spinlocks used, so we can sort of kind of safely ignore the interrupt state. There might be a better solution with another boot state flag that checks whether APs are up (because they have interrupts enabled from the start) but that seems overkill.

This PR makes the Kernel progress back into init stage 2 with LOCK_CRITICAL_DEBUG, a flag we (eventually) want to remove and have the checks always run. This fixes regressions from a couple of months ago; @ADKaster reported that we already had init stage 2 with this flag at some point.

### CODE REVIEW

File: Kernel/Locking/Mutex.cpp [hunk 1]
```diff
@@ -12,15 +12,19 @@
...
 {
     // NOTE: This may be called from an interrupt handler (not an IRQ handler)
     // and also from within critical sections!
     VERIFY(!Processor::current_in_irq());
-    if constexpr (LOCK_IN_CRITICAL_DEBUG)
-        VERIFY_INTERRUPTS_ENABLED();
+    // There are no interrupts enabled in early boot.
+    if (!g_in_early_boot)
+        if constexpr (LOCK_IN_CRITICAL_DEBUG)
+            VERIFY_INTERRUPTS_ENABLED();
```
Comment (GMTA) [inline Kernel/Locking/Mutex.cpp:27]: 1. Please add braces around an `if` block with more than one line
2. I'd switch the `constexpr` and `g_in_early_boot` lines: that way it's clear that both lines are guarded by the debug flag (which eventually needs to be removed)

File: Kernel/Locking/Mutex.cpp [hunk 2]
```diff
@@ -12,15 +12,20 @@
...
 namespace Kernel {
 
 void Mutex::lock(Mode mode, [[maybe_unused]] LockLocation const& location)
 {
     // NOTE: This may be called from an interrupt handler (not an IRQ handler)
     // and also from within critical sections!
     VERIFY(!Processor::current_in_irq());
-    if constexpr (LOCK_IN_CRITICAL_DEBUG)
-        VERIFY_INTERRUPTS_ENABLED();
+    // There are no interrupts enabled in early boot.
```
Comment (linusg) [inline Kernel/Locking/Mutex.cpp:24]: This comment should go before the second if IMO

File: Kernel/Locking/Mutex.cpp [hunk 3]
```diff
@@ -147,9 +152,12 @@ void Mutex::unlock()
 {
     // NOTE: This may be called from an interrupt handler (not an IRQ handler)
     // and also from within critical sections!
-    if constexpr (LOCK_IN_CRITICAL_DEBUG)
-        VERIFY_INTERRUPTS_ENABLED();
     VERIFY(!Processor::current_in_irq());
+    // There are no interrupts enabled in early boot.
```
Comment (linusg) [inline Kernel/Locking/Mutex.cpp:156]: Likewise

File: Kernel/Locking/Mutex.cpp [hunk 4]
```diff
@@ -205,8 +213,11 @@ void Mutex::unlock()
 
 void Mutex::block(Thread& current_thread, Mode mode, SpinlockLocker<Spinlock>& lock, u32 requested_locks)
 {
-    if constexpr (LOCK_IN_CRITICAL_DEBUG)
-        VERIFY_INTERRUPTS_ENABLED();
+    // There are no interrupts enabled in early boot.
```
Comment (linusg) [inline Kernel/Locking/Mutex.cpp:216]: Likewise

**Pull request merged**